lib/pages/carteiraApp.dart

```diff
@@ -1,118 +1,150 @@
 import 'package:flutter/cupertino.dart';
 import 'package:flutter/material.dart';
 
+import '../models/coin.dart';
 import '../repositories/coin_repository.dart';
+import 'moedas_detalhes_page.dart';
 
 class TelaInicialWidget extends StatefulWidget {
-
   const TelaInicialWidget({Key? key}) : super(key: key);
 
   @override
   State<TelaInicialWidget> createState() => TelaInicialState();
-
 }
 
 class TelaInicialState extends State<TelaInicialWidget> {
+  final tabela = MoedaRepository.tabela;
 
-    final tabela = MoedaRepository.tabela;
-    
-    bool opacidade = true;
-    String carteira = 'Carteira';
-    
-    Widget listagemCrypto(int moeda){
-         return
-             Column(
-                children: <Widget>[
-                  Container(
-                      height: 0,
-                  ),
-                  ListTile(
-                       leading: SizedBox( width: 30, child: Image.asset(tabela[moeda].icone),),
-                       title: Text(tabela[moeda].sigla),
-                       subtitle: Text(tabela[moeda].moeda),
-                      trailing: AnimatedOpacity(
-                        opacity: opacidade ? 1 : 0,
-                        duration: const Duration(milliseconds: 800),
-                        child: Column(
-                          children: [
-                            Text(tabela[moeda].valor),
-                            Text('${tabela[moeda].variacao}%',
-                              style: TextStyle(fontSize: 10, fontWeight: FontWeight.bold, color: (tabela[moeda].variacao > 0) ? Colors.green : Colors.red,
-                              backgroundColor: (tabela[moeda].variacao > 0) ? Colors.green[50] : Colors.red[50]
-                            ),)
-                          ],
-                        ),
-                      ),
-                   
-                  )
-                ],
-         );
-    }
+  bool opacidade = true;
+  String carteira = 'Carteira';
+  List<Moeda> selecionadas = [];
+  Widget listagemCrypto(int moeda) {
```

Comment:
Não é uma boa prática criar uma função que retorna um Widget. O recomendável é criar outro Widget, pode ser um privado no mesmo arquivo se for somente desse escopo.

https://dartcodemetrics.dev/docs/rules/flutter/avoid-returning-widgets

Reply:
Ajustado , implementado um outro Widget no para substituir a função que retornava um Widget